src/auth/useUser.ts

```diff
@@ -12,17 +12,18 @@ import {
 } from "firebase/auth";
 import { useEffect, useState } from "react";
 
-type UseAuthStateParams = {
-  auth: Auth;
+type UseAuthStateOptions = {
   onError?: ErrorFn;
   onChange?: CompleteFn;
 };
 
-export const useUser = ({
-  auth,
-  onError,
-  onChange,
-}: UseAuthStateParams): User | null => {
+export const useUser = (
+  auth: Auth,
+  { onError, onChange }: UseAuthStateOptions = {
+    onError: () => {},
+    onChange: () => {},
+  },
+): User | null => {
   const [user, setUser] = useState<User | null>(auth.currentUser);
 
   useEffect(() => {
```

Comment:
Maybe the error occurs due to `onAuthStateChanged` does not drop resources or is not okay with concurrent usage.

Reply:
The error was due to incorrect use of parameters.